openembedded-core-mbl/scripts/signctl/README.md

```diff
@@ -0,0 +1,329 @@
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --rpi-armstub8 RPI_ARMSTUB8
+                        Path to armstub8.bin
+  --fip FIP [FIP ...]   Path to one or more FIPs to be signed.
+  --patch-rotkey        Patch root of trust public key hash in to BL1 and BL2.
+  --output-dir OUTPUT_DIR
+                        Output directory for signed images and the original
+                        fip components.
+```
+
+## Generating Chain of Trust Keychain
+
+You can generate the keychain used for the ATF CoT using the `signctl generate --cot-keys` command.
+
+Currently the `generate --cot-keys` command creates all of the required CAs up to the root of trust. The private keys are stored in the Vault.
+
+In the release version, the `signctl` tool will support generating individual keypairs in the keychain, allowing for key revocation and access restriction to high security keys.
+
+<span class="notes">**Note:** The tool only supports signing for a single device at the moment. The release version will support multiple devices; we will provide documentation and guidance for setting access control policies in Vault, and for locking down the Vault machine for production use. **The current flow is only for development use.**</span>
+
+To use the generate command follow these steps:
+
+1. Ensure your Vault server is running. To run a local dev server use the Vault CLI, `vault server -dev`.
+2. Add a file named `.signctl.conf` in your user's home directory. This file should contain only the URL and port of the Vault server, in the form `url:port`. Alternatively you can pass the url and port of the Vault server on the command line, when running the `generate` command, by adding the `--backend-url URL` flag.
+3. Run the following command
+```
+$ signctl generate --cot-keys
+```
+
+<span class="notes">**Note:** If using the `--backend-url` flag, it should be entered before the command. For example:
+
+```
+$ signctl --backend-url 127.0.0.1:8200 generate --cot-keys
+```
+</span>
+
+The keys should be shown as generated in the Vault server output. If the `signctl` tool returns with no output it means the keys were successfully generated.
+
+## Signing RPi3 Boot Components
+
+After the keys have been generated and stored in Vault, you can use them to sign the certificates needed to verify the FIP.
+
+To produce the signed leaf certificates and pack them into a FIP use the `signctl sign-atf` command.
+
+If you saved your Vault server URL in a `.signctl.conf` file as described in the previous section, the `signctl` tool will automatically connect to the Vault server. If not you will have to pass the server URL again on the command line.
+
+You can also patch the root of trust public keys into BL1 and BL2 by using the `--patch-rotkeys` command. The background for this is described in the [the basic signing flow document](https://github.com/ARMmbed/meta-mbl/blob/warrior-dev/docs/basic-signing-flow.md).
```

Comment:
As above use indirect links and relative links for docs.

Reply:
Added indirect links in bd16b66